### ISSUE TITLE
micro-ROS Agent on Raspberry 5 with /dev/AMA0 does not work

### ISSUE THREAD

**Issue by mucknem1** [micro-ROS/micro-ROS-Agent#228]
Hello,

I have setup a micro ROS Agent on a Raspberry PI 5 and connected to microROS client with UART. The UART on the PI is connected to Pin14/Pin15 (Rx/Tx) with /dev/AMA0. Which does not work. I get no connection to client.

If I connect a FTDI USB-UART serial adapter the USB port and connect the agent to port /dev/ttyUSB0 it does work as expected. 

The UART AMA0 connection to my PC does work, which I have tested with the minicom terminal and hTerm

This is my setup
- Hardware description: Raspberry Pi 5
- RTOS: Ubuntu 24.04 
- Installation type: micro-ROS Agent
- Version or commit hash: micro-ROS iron

Does anyone have an idea why it does not work with the onboard UART. Are there any special settings I need to make?

many thanks in advanced

**Comment by dvdtrk** [comment]
Yep, same issue on a Pi 5 (Ubuntu, ROS 2 Kilted) with a Pico client on /dev/ttyAMA2. Raw serial loopback works fine (verified with minicom + a hex dump of the client's TX frames), but the agent never logs anything past running.../logger setup even at max verbosity, and the client's ping never succeeds. Both directions confirmed working at the raw serial level, so it seems like the handshake itself is failing specifically over onboard UART. I would like to see some updates on this.
